### ISSUE TITLE
Logic for determining whether something is an "id" field is incorrect

### ISSUE THREAD

**Issue by goldcaddy77** [goldcaddy77/warthog#208]
In the event where you have fields that end in `Id` that are not IDs, they get the wrong set of filters:

- `userId` - is an email address
- `externalId` - could be something else that should be searchable like a string

We should make sure to only give the ID treatment to ID fields.

**Comment by goldcaddy77** [comment]
Be smarter here: https://github.com/goldcaddy77/warthog/blob/master/src/schema/TypeORMConverter.ts#L55-L57

**Comment by nwmartin** [comment]
Would this result in some generation behavior like this?

```
@Model()
export class Order extends BaseModel {
  @IntField({ nullable: false })
  agreementId!: number;
...
}

@TypeGraphQLInputType()
export class OrderCreateInput {
  @TypeGraphQLField()
  agreementId!: string;
...
}
```

I may have just run into this. Why do we need to ensure a GraphQLID type for all columns ending in ID exactly? Couldn't this be more manually specific? I might be missing something here.

**Comment by goldcaddy77** [comment]
Yes, this is incorrect. We should remove the magic around ID fields getting their ID treatment from the field name. It should come from using a specific decorator or being a OneToMany/ManyToOne

**Comment by goldcaddy77** [comment]
@nwmartin It's hard to fix this without a breaking change, but I put in a fix here: https://github.com/goldcaddy77/warthog/pull/232/files#diff-26d96fc192e5df2bfc3a94e5a79c43d0R24-R40 .  It requires adding an explicit filter type when you have a field that ends in `Id`.  This will be fixed in the move to v2.0.

**Comment by goldcaddy77** [comment]
Fixed in version 2.0.0-beta.8.  Closing
